calcom/cal.com
# Bookings with no-show fee to set proper location & prevent broken payments from blocking payment page

## What does this PR do?



This PR fixes a bug in which locations were not respected when bookings enabled no-show fees. Also if a payment record was malformed, it would block the bookings page from loading

Fixes # (issue)



**Environment**: Staging(main branch) / Production

## Type of change



- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?



- Create an event type with a location other than Cal video & Stripe enabled set to charge a no-show fee
- Create a booking
- After entering your payment information the confirmed booking emails should contain the set location link
- In the DB delete the payment record for that booking
- Go to the bookings page. It should still load with an error message
    
![image](https://user-images.githubusercontent.com/65426560/234171367-a4ff32bb-cf76-490e-b76b-a9b912edc653.png)

## Checklist


- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
